### PR TITLE
Integration test suite update

### PR DESCRIPTION
**Pull request by hannydevelop**
This PR is a work in progress. We're moving tests from the `testnet` module into the `integration_test` suite. The tests from `testnet` module are outdated and don't work properly. We're refining these tests for gravity-bridge to make sure they work as expected in the `integration_test` suite

### CODE REVIEW

File: integration_tests/valset_stress_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// call stress_test for a range of nonce values
+
+/// Write test_valset_update test to get latest nonce value
+func (s *IntegrationTestSuite) TestValsetUpdate(val sdk.ValAddress) {
```
Comment (mvid) [inline integration_tests/valset_stress_test.go:17]: Is this `val` used anywhere?

File: integration_tests/valset_stress_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	s.Run("Bring up chain, and test the valset update", func() {
+		ethClient, err := ethclient.Dial(fmt.Sprintf("http://%s", s.ethResource.GetHostPort("8545/tcp")))
+		s.Require().NoError(err, "error setting up eth client")
+
+		starting_nonce, err := ethClient.PendingNonceAt(context.Background(), gravityContract)
```
Comment (mvid) [inline integration_tests/valset_stress_test.go:22]: Go convention is camelCase vs snake_case

File: integration_tests/valset_stress_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		} else {
+			s.T().Logf("Failed to update validator set")
+		}
+
+	})
+}
+
+// loop and test valset update from a range of nonces
+func (s *IntegrationTestSuite) TestValsetUpdate() {
+	s.Run("Bring up chain, and test the valset update", func() {
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply integration_tests/valset_stress_test.go:112]: No, we just loop one integration test 10 times, to get 10 nonces. Here's a peep of what it looks like.

```
=== RUN   TestIntegrationTestSuite
    setup_test.go:83: setting up e2e integration test suite...
    setup_test.go:89: starting e2e infrastructure; chain-id: chain-o7kS3S; datadir: /var/folders/ns/20wg_vm169s09pcp0m36vqhw0000gn/T/gravity-bridge-e2e-testnet2999010844
    setup_test.go:423: starting Ethereum container...
    setup_test.go:458: error querying balance: &{%!e(string=Post) %!e(string=http://localhost:8545) %!e(*errors.errorString=&{EOF})}
    setup_test.go:478: waiting for contract to deploy
    setup_test.go:497: gravity contract deployed at 0x04C89607413713Ec9775E14b954286519d836FEf
    setup_test.go:509: test erc20 contract deployed at 0x4C4a2f8c81640e47606d3fd77B353E87Ba015584
    setup_test.go:511: started Ethereum container: da4ffa4723b6afae705cc858920740df8364e2caa4a3de6ee87e134767ccd19f
    setup_test.go:515: starting validator containers...
    setup_test.go:545: started validator container: 5e25d2f1a520ad8875d998b467c4fcec556746853a73ec38df78591f7af530b6
    setup_test.go:545: started validator container: af21892e635b7b993e3e2e4e7189c76ab20b6d82e47309b7d63f855d155d8ee7
    setup_test.go:545: started validator container: 1b75d5b6f7b20ae36bc07af96b0dc580afa2397082be19cb829ec3d0e8593b09
    setup_test.go:545: started validator container: 16661a8d93a4fca8dc4d5d68298b47ac6e5cc53dd5f91a6dd519fc4dae47e893
    setup_test.go:590: starting orchestrator containers...
    setup_test.go:669: started orchestrator container: 3105278a13d5f99fa6dea448c6c3682d8f0e92ae256463cefab49a8603793d1e
    setup_test.go:669: started orchestrator container: db211051d801cfbaaf556b89beb58b196711c774f39a81075c70581175912bfb
    setup_test.go:669: started orchestrator container: 0f1c627aa8a921f605882b04a91e2574c82286d5f9df9e6b710f5de8fc6fcd80
    setup_test.go:669: started orchestrator container: 838335146d3e2a7dd97bf652c712c86cac36fabb427ae001b5933e2c5ba5208b
    setup_test.go:678: waiting for orchestrator to be healthy: 3105278a13d5f99fa6dea448c6c3682d8f0e92ae256463cefab49a8603793d1e
    setup_test.go:678: waiting for orchestrator to be healthy: db211051d801cfbaaf556b89beb58b196711c774f39a81075c70581175912bfb
    setup_test.go:678: waiting for orchestrator to be healthy: 0f1c627aa8a921f605882b04a91e2574c82286d5f9df9e6b710f5de8fc6fcd80
    setup_test.go:678: waiting for orchestrator to be healthy: 838335146d3e2a7dd97bf652c712c86cac36fabb427ae001b5933e2c5ba5208b
=== RUN   TestIntegrationTestSuite/TestValsetUpdate
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374404743968})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "50000000000.000000000000000000"
        
          Balance:   50000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:2 height:78 signers:<power:1431655765 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:954437176 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:954437176 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:954437176 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374395213376}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#01
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374404517984})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "100000000000.000000000000000000"
        
          Balance:   100000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:3 height:94 signers:<power:1717986918 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:858993459 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:858993459 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:858993459 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374395214048}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#02
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374395218336})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "150000000000.000000000000000000"
        
          Balance:   150000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:4 height:111 signers:<power:1952257861 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:780903144 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:780903144 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:780903144 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374409883808}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#03
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374417310688})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "200000000000.000000000000000000"
        
          Balance:   200000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:5 height:128 signers:<power:2147483647 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:715827882 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:715827882 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:715827882 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374417312192}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#04
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374404520160})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "250000000000.000000000000000000"
        
          Balance:   250000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:6 height:145 signers:<power:2312674697 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:660764199 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:660764199 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:660764199 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374417306816}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#05
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374389732608})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374389732608})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "300000000000.000000000000000000"
        
          Balance:   300000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:7 height:169 signers:<power:2454267025 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:613566756 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:613566756 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:613566756 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374417313024}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#06
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374409889728})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "350000000000.000000000000000000"
        
          Balance:   350000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:8 height:186 signers:<power:2576980377 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:572662306 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:572662306 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:572662306 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374417313568}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#07
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374389738528})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "400000000000.000000000000000000"
        
          Balance:   400000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:9 height:204 signers:<power:2684354559 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:536870911 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:536870911 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:536870911 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374404739776}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#08
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374389738720})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "450000000000.000000000000000000"
        
          Balance:   450000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:9 height:204 signers:<power:2684354559 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:536870911 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:536870911 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:536870911 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374389739232}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#09
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374395213440})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "500000000000.000000000000000000"
        
          Balance:   500000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:10 height:238 signers:<power:2863311530 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:477218588 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:477218588 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:477218588 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374404739968}
=== RUN   TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#10
    valset_stress_test.go:35: Sending in valset request (starting_eth_valset_nonce &{1374389737568})
    valset_stress_test.go:37: Delegating 50000000000testgb to 431CAA1ECF1377E9559BFCC17F0EAEA6C16FB56D in order to generate a validator set update
    valset_stress_test.go:56: verifying delegation
    valset_stress_test.go:66: Here's the delegation response: delegator_address: cosmos1gvw258k0zdm7j4vmlnqh7r4w5mqkldtda8wkkl
        validator_address: cosmosvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e59lynph
        shares: "550000000000.000000000000000000"
        
          Balance:   550000000000testgb
    valset_stress_test.go:71: verifying signerset
    valset_stress_test.go:81: Here's the last signerset response: nonce:10 height:238 signers:<power:2863311530 ethereum_address:"0x14fdAC734De10065093C4Ed4a83C41638378005A" > signers:<power:477218588 ethereum_address:"0x7bE2a04df4b9C3227928147461e19158eB2B11d1" > signers:<power:477218588 ethereum_address:"0xb8c6886FDDa38adaa0F416722dd5554886C43055" > signers:<power:477218588 ethereum_address:"0xd312f0f1B39D54Db2829537595fC1167B14d4b34" > 
    valset_stress_test.go:101: Validator set successfully updated! nonce: &{1374395212992}
=== CONT  TestIntegrationTestSuite
    setup_test.go:121: tearing down e2e integration test suite...
--- PASS: TestIntegrationTestSuite (411.24s)
    --- PASS: TestIntegrationTestSuite/TestValsetUpdate (268.64s)
        --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update (268.64s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update (24.15s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#01 (22.76s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#02 (23.77s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#03 (23.65s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#04 (23.89s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#05 (33.01s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#06 (24.20s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#07 (23.96s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#08 (23.00s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#09 (22.62s)
            --- PASS: TestIntegrationTestSuite/TestValsetUpdate/Bring_up_chain,_and_test_the_valset_update/Bring_up_chain,_and_test_the_valset_update#10 (23.62s)
PASS
```

File: Makefile [hunk 4]
```diff
@@ -65,3 +65,6 @@ fail:
 
 e2e_happy_path: e2e_clean_slate
 	@E2E_SKIP_CLEANUP=true integration_tests/integration_tests.test -test.failfast -test.v -test.run IntegrationTestSuite -testify.m TestHappyPath || make -s fail
+
+e2e_valset_update: e2e_clean_slate
+	@E2E_SKIP_CLEANUP=true integration_tests/integration_tests.test -test.failfast -test.v -test.run IntegrationTestSuite -testify.m TestValsetUpdate || make -s fail
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply Makefile:70]: done!

File: integration_tests/valset_stress_test.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		// Run a loop until you get a nonce higher than the initial nonce
+		s.Require().Eventuallyf(func() bool {
+			for currentNonce == startingNonce {
+				currentNonce, err = s.getLastValsetNonce(gravityContract)
+				if currentNonce != startingNonce {
+					return true
+				}
+			}
+			return true
+		}, 300*time.Second, 10*time.Second, "Validator set is not yet updated")
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply integration_tests/valset_stress_test.go:98]: great, done

**Pull request merged**